x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/MustacheTemplateEvaluator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.security.support;
+
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptService;
+import org.elasticsearch.script.ScriptType;
+import org.elasticsearch.script.TemplateScript;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for evaluating Mustache templates at runtime.
+ */
+public final class MustacheTemplateEvaluator {
+
+    private MustacheTemplateEvaluator() {
+        throw new UnsupportedOperationException("Cannot construct " + MustacheTemplateEvaluator.class);
+    }
+
+    public static String evaluate(ScriptService scriptService, XContentParser parser, Map<String, Object> extraParams) throws IOException {
```

Comment:
We should disallow triple braces (`{{{`) or `{{&` syntax as it bypasses escaping and we have no use for it. This can potentially be handled in the ScriptService for all the codebase in a separate change, so I'm adding it here as a mental note, not as _necessarily_ something that should be tackled in this PR

Reply:
I think we should do this in the script engine rather than having special cases here.